Lib/test/datetimetester.py

```diff
@@ -1795,6 +1795,75 @@ def test_fromisoformat_fails_typeerror(self):
             with self.assertRaises(TypeError):
                 self.theclass.fromisoformat(bad_type)
 
+    def test_fromisocalendar(self):
+        # For each test case, assert that fromisocalendar is the
+        # inverse of the isocalendar function
+        dates = [
+            (2016, 4, 3),
+            (2005, 1, 2),       # (2004, 53, 7)
+            (2008, 12, 30),     # (2009, 1, 2)
+            (2010, 1, 2),       # (2009, 53, 6)
+            (2009, 12, 31),     # (2009, 53, 4)
+            (1900, 1, 1),       # Unusual non-leap year (year % 100 == 0)
+            (1900, 12, 31),
+            (2000, 1, 1),       # Unusual leap year (year % 400 == 0)
+            (2000, 12, 31),
+            (2004, 1, 1),       # Leap year
+            (2004, 12, 31),
+            (1, 1, 1),
+            (9999, 12, 31),
```

Comment:
Maybe use MINYEAR/MAXYEAR here.

Reply:
Instead of converting these over, I have just added a `MINYEAR` and `MAXYEAR` test. Even though it's redundant, I like to have the explicit callout of the current boundaries, so that if `MINYEAR` or `MAXYEAR` get modified in a way that breaks backwards compatibility, it will raise an error.